### ISSUE TITLE
[FP] Null pointer exeption after ?:

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1221]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate) v2.0.12 |
| Language level | PHP 5.6 |

Current behaviour
--
if you have a var that may be a class or null, and you then use ?: to make sure its not null,
the inspection still wans that it may be null:
![screenshot from 2019-03-05 10-17-16](https://user-images.githubusercontent.com/866668/53794256-ea64d400-3f2f-11e9-8fad-38944c85123f.png)

```php
<?php

	class Foo
	{
		public $bar;

		/**
		 * @return Foo|null
		 */
		public static function maybe()
		{
			static $c;
			if($c++ & 1)
			{
				return NULL;
			}
			return new self();
		}

		public static function test()
		{
			$a = self::maybe();
			$b = $a ? clone $a : new self();
			$b->bar = 1;
			$c = /*$a ? clone $a :*/ new self();
			$c->bar = 1;
		}
	}
```

Expected behaviour
--
Just as `$c` can't be null, `$b` can't be null.
(also tried to removed the clone, and it still gave a warning)

Environment details
---
PhpStorm 2019.1 EAP
Build #PS-191.5532.18, built on February 13, 2019
PhpStorm EAP User
JRE: 1.8.0_202-release-1483-b4 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.18.0-15-generic

**Comment by kalessil** [comment]
Fixed!
